ui/spec/dashboards/reducers/uiSpec.js

```diff
@@ -50,6 +54,25 @@ describe('DataExplorer.Reducers.UI', () => {
     expect(actual.dashboard).to.deep.equal(d2)
   })
 
+  it('can handle a successful dashboard deletion', () => {
+    const loadedState = reducer(state, loadDashboards(dashboards))
+    const expected = [d1]
+    const actual = reducer(loadedState, deleteDashboard(d2))
+
+    expect(actual.dashboards).to.deep.equal(expected)
+  })
+
+  it('can handle a failed dashboard deletion', () => {
+    const loadedState = reducer(state, loadDashboards([d1]))
+    const actual = reducer(loadedState, deleteDashboardFailed(d2))
+    const actualFirst = _.first(actual.dashboards)
+
+    expect(actual.dashboards).to.have.length(2)
```

Comment:
We have to remember as we're writing these tests that redux reducers must treat prior state as immutable. If we're going through mental gymnastics to test our code, not only is that a possible code smell, that could be indicative of a mutability issue.